extension/src/emulator/tools/account-manager.ts

```diff
@@ -0,0 +1,124 @@
+/*  AccountManager is responsible for creating and switching accounts.
+    It deals with the local configuration and language server.
+*/
+import { ext } from '../../main'
+import { AccountData } from '../local/account-data'
+import { LanguageServerAPI } from '../server/language-server'
+import { window, env } from 'vscode'
+import { Account } from '../account'
+import {
+  COPY_ADDRESS,
+  CREATE_NEW_ACCOUNT,
+  ACTIVE_PREFIX,
+  INACTIVE_PREFIX,
+  ADD_NEW_PREFIX
+} from '../../utils/strings'
+
+export class AccountManager {
+  api: LanguageServerAPI
+  accountData: AccountData
+
+  constructor (api: LanguageServerAPI) {
+    this.api = api
+    this.accountData = new AccountData()
+  }
+
+  async addAccountLocal (account: Account): Promise<void> {
+    this.accountData.addAccount(account)
+  }
+
+  async createNewAccount (): Promise<void> {
+    try {
+      const account = await this.api.createAccount()
+      this.accountData.addAccount(account)
+      const lastIndex = this.accountData.accounts.length - 1
+      await this.setActiveAccount(lastIndex)
```

Comment:
```suggestion
      const index = this.accountData.addAccount(account)
      this.setActiveAccount(index)
```

Something like this might be nice. Feels like `this.accountData.accounts.length - 1` requires a bit too much knowledge of the internal `AccountData` implementation.

Reply:
Sounds good!